Sources/StreamChat/StateLayer/ChatState+Observer.swift

```diff
@@ -38,8 +39,8 @@ extension ChatState {
                     for: cid,
                     pageSize: channelQuery.pagination?.pageSize ?? .messagesPageSize,
                     sortAscending: messageOrder.isAscending,
-                    deletedMessagesVisibility: context.deletedMessagesVisibility ?? .visibleForCurrentUser,
-                    shouldShowShadowedMessages: context.shouldShowShadowedMessages ?? false
+                    deletedMessagesVisibility: clientConfig.deletedMessagesVisibility,
+                    shouldShowShadowedMessages: clientConfig.shouldShowShadowedMessages
```

Comment:
This was throwing an assert in Apple's code since I was accessing `NSManagedObjectContext`'s `userInfo` from the main thread when running tests. Realised that it is nicer to use the config directly since then I do not need to think about the default value.